index.html

```diff
@@ -118,17 +238,75 @@
     // TRANSFORM
     txData = data.slice();
 
-    if (options.yearRange !== 'all') {
-      var yearRange = options.yearRange.split('-');
+
+    if (options.state !== 'all') {
       txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
-      });
+        return d.state === options.state;
+      }); 
     }
 
     // SCATTERPLOT
 
+
+    var points = svg.selectAll('.point')
+                    .data(txData);
+
+                    points.enter().append('circle')
+                    .attr('class', 'point')
+                    .attr('r', 5);
+
+                    points
+                    .attr('cx', function(d) { return x(d.democratContribution); })
+                    .attr('cy', function(d) { return y(d.republicanContribution)});
+
+                    points.exit().remove();
+
     // HISTOGRAMS
 
+    var barWidth = ( x(x.ticks()[1]) - x(x.ticks()[0]) )/5;
+
+    // x axis histogram
+    var demContr = txData.map(function (i) {return i.democratContribution});
+
+    var xHistogram = d3.layout.histogram()
+                             .bins(x.ticks())
+                             (demContr); 
+
+    histScale = d3.scale.linear()
+                    .domain([0, d3.max(xHistogram, function (d) { return d.y; })])
+                    .range([0, height/5])
+                    .nice();
+
+    var xBars = svg.selectAll(".xbar")
+                  .data(xHistogram)
+                  .enter().append("g")
+                  .attr("class", "xbar");
+                  
+    xBars.attr("transform", function(d, i) { return "translate(" + ( ((x(x.ticks()[i+1]) - (x(x.ticks()[i])))/2) + x(x.ticks()[i]) - (barWidth/2) )  + ",0)"; })
```

Comment:
💯 Nice work with the math and dealing with the logarithmic scale here. This works great. Now that we've reviewed the `x` and `dx` that we get from the histogram layout, perhaps you can see some ways for this to be tidied up (I'd also love to get a line break after the curly brace, just so the line doesn't get too long). But yeah, this'll do it.